### PR TITLE
Header profile

### PR DESCRIPTION
**Pull request by NadaJrad**
this pull request  have problems i need to pankaj review it first

### CODE REVIEW

File: app/app/ .env.local [hunk 1]
```diff
@@ -0,0 +1,2 @@
+
+NEXT_PUBLIC_API_BASE_URL=http://localhost:3001
```
Comment (pankaj28843) [inline app/app/ .env.local]: add this to .gitignore file and add a `.env.local.example` instead which can be on Github.

File: app/app/components/Header.jsx [hunk 2]
```diff
@@ -1,63 +1,94 @@
 "use client";
-
-import React, { useState } from "react";
+import React, { useState, useEffect } from "react";
 import Image from "next/image";
 import styles from "./Header.module.css";
 import Link from "next/link";
 import { useRouter } from "next/navigation";
+import Cookies from "js-cookie";
+import { jwtDecode } from "jwt-decode"; // Correct import
```
Comment (tanya-newStart) [inline app/app/components/Header.jsx]: please use useAuth() here, no need for token in the code in this file

File: app/app/ .env.local (hunk 1 above)
Comment (pankaj28843) [reply app/app/ .env.local]: This file is at the wrong location, it should be at `app/.env.local`

File: app/app/components/Header.jsx [hunk 3]
```diff
@@ -1,63 +1,97 @@
...
+  const { currentUser, logout } = useAuth(); // Use currentUser and logout from AuthContext
+
+  // Conditional greeting message: Show username if logged in, or 'Guest' if not
+  const userName = useMemo(
+    () => currentUser?.user.firstName || "Guest",
+    [currentUser]
+  );
+
+  // For debugging purposes
+  console.log(`currentUser: ${JSON.stringify(currentUser, null, 2)}`);
```
Comment (magdazelena) [inline app/app/components/Header.jsx:22]: you should delete the console logs upon creating the PR :)

File: app/app/components/Header.jsx [hunk 4]
```diff
@@ -1,63 +1,97 @@
...
+  const handleSearchKeyPress = (e) => {
+    if (e.key === "Enter" && searchQuery.trim() !== "") {
+      router.push(`/search?q=${encodeURIComponent(searchQuery.trim())}`);
+    }
+  };
+
+  // Handle logout functionality
+  const handleLogout = async () => {
+    try {
+      await logout(); // Call the logout function from AuthContext
```
Comment (magdazelena) [inline app/app/components/Header.jsx:39]: same goes to comments - code should be self-explanatory. it is not wrong, but general approach is to create the variables and functions that speak for themselves

File: app/app/components/Header.jsx [hunk 5]
```diff
@@ -1,63 +1,97 @@
...
+      router.push("/login"); // Redirect to the login page after logging out
+    } catch (error) {
+      console.error("Logout failed:", error.message);
```
Comment (magdazelena) [inline app/app/components/Header.jsx:42]: usually after a failed action some sort of feedback to the user is necessary like a popup or banner with clear information and call to action

**Review by magdazelena — COMMENTED** [review]
some comments :)

**Pull request closed without merging**